drivers/ieee802154/ieee802154_nrf5.h

```diff
@@ -89,6 +89,39 @@ struct nrf5_802154_data {
 
 	/* Indicates if currently processed TX frame has dynamic data updated. */
 	bool tx_frame_mac_hdr_rdy;
+
+#if defined(CONFIG_IEEE802154_NRF5_MULTIPLE_CCA)
+	/* The maximum number of extra CCA attempts to be performed before transmission. */
+	uint8_t extra_cca_attempts;
+#endif
 };
 
+#if defined(CONFIG_IEEE802154_NRF5_MULTIPLE_CCA)
+/**
+ * @brief Sets the maximum number of extra CCA attempts to be performed by a transmit operation
+ *
+ * The default value of extra cca attempts is 0.
+ *
+ * The maximum number of extra cca attempts set by this function is applied to transmissions
+ * requested with mode IEEE802154_TX_MODE_TXTIME_CCA only. This might change in the future, so
+ * it is recommended to restore previously used value after each transmission. See
+ * @ref z_ieee802154_nrf5_extra_cca_attempts_get.
+ *
+ * @param dev    Pointer to a ieee802154_nrf5 device
+ * @param value  Value to set. Allowed range is 0...254.
+ */
+void z_ieee802154_nrf5_extra_cca_attempts_set(const struct device *dev, uint8_t value);
```

Comment:
Is my understanding correct that in order to use this functionality, vendor-agnostic code that usually operates on generic, vendor-agnostic ieee802154 and net APIs would have to call this function directly?

Reply:
See https://github.com/zephyrproject-rtos/zephyr/pull/60390#issuecomment-1635701165

Reply:
@ankuns That's the kind of indirection that I think is hard to maintain/read and keep race-condition free as you'd have to place locking around the whole call site I guess - see my comment below.